src/Development/IDE/Core/Rules.hs

```diff
@@ -784,6 +785,12 @@ isFileOfInterestRule = defineEarlyCutoff $ \IsFileOfInterest f -> do
     let res = f `elem` filesOfInterest
     return (Just (if res then "1" else ""), ([], Just res))
 
+
+getClientSettingsRule :: Rules ()
+getClientSettingsRule = defineNoFile $ \GetClientSettings -> do
+  alwaysRerun
```

Comment:
An `alwaysRerun` rule needs early cutoff - please use `defineEarlyCutoffNoFile`. 
Moreover, following my advice in #688, please change the `RuleResult GetClientSettings` to `Hashed (Maybe Value)`

Reply:
This isn't fully resolved, please change the RuleResult GetClientSettings to Hashed (Maybe Value)

Reply:
yeah, I was about to change it in another commit, is it for not recomputing the hash every time or is there another reason?

Reply:
that's right

Reply:
I would like to keep `clientSettings IdeConfiguration` with type `Maybe Value`, to not leak the performance related wrapper to client code, would be it possible? (i guess it is not, after taking a look to #688)

Reply:
nvm, i am happy just keeping `getClientSettings` and `modifyClientSettings` with the same signature, without `Hashed`